### ISSUE TITLE
Resolve Text Input Code Divergence

### ISSUE THREAD

**Issue by InMatrix** [flutter/flutter#122647]
I was looking to use `CupertinoTextField` on iOS and `TextField` on Android for my app. The API differences between the two widgets for seemingly similar concepts feel rather unnecessary. Here are some examples:

- To add a piece of hint text, I need to supply that through the `decoration` property on `TextField` but the `placeholder` property on `CupertinoTextField`. 
- The `decoration` property on `TextField` takes an `InputDecoration`, but the same property on `CupertinoTextField` takes a `BoxDecoration`.
- To add a prefix icon, I can use the `prefixIcon` property on `InputDecoration`, which I then pass to the `decoration` property on `TextField`. However, I need to use the `prefix` property on `CupertinoTextField` to achieve a similar result. 

Such differences make it difficult to 1) reason about how to use `CupertinoTextField` based on my knowledge about `TextField`, and 2) minimize platform-specific code when showing one of the two widgets based on the OS. 

It could be helpful if a base text field is provided and design language is applied through theming or decoration.

**Comment by HansMuller** [comment]
CC @justinmc @MitchellGoodwin 
This is an impediment to writing portable code that's definitely worth investigating. Creating support for `TextField.adaptive()` would be non-trivial but perpetuating an incompatible API for Cupertino textfields creates an obstacle for many cross-platform apps. We will investigate.

**Comment by InMatrix** [comment]
Thanks for taking a look @HansMuller. 

Here's some food for thought. I was able to customize the M3 TextField widget into something quite close to `CupertinoTextField` in appearance. Here's a demo in DartPad: https://dartpad.dev/?id=5e21d29ef2dc94ddad0a4a635679cfad. As an app developer, I'm not sure if I'd choose this approach over using `CupertinoTextField`. As it stands today, I'd be writing platform specific code either way. However, if this kind of customization came with the Material library, then it can provide some convenience.

**Comment by leighajarett** [comment]
I think this has become an even bigger issue with the spellcheck work. On iOS, TextField is not really usable with spellcheck, because of the way the word suggestions are displayed. It's very different from the expected behavior and when I tested it, it overflows. 
![Apr-07-2023 13-20-18](https://user-images.githubusercontent.com/21065911/230650530-e89c64ed-a6fe-4098-8f09-2ff315047dd1.gif)

**Comment by justinmc** [comment]
@camsim99 Is there a GitHub issue for the bug posted by @leighajarett above?

**Comment by justinmc** [comment]
I'm super interested in this greater idea by @InMatrix to build iOS-looking widgets in Material.  I'm definitely on board with helping to support that and/or trying to synchronize the APIs for TextField and CupertinoTextField more.

**Comment by cpboyd** [comment]
I agree with @justinmc. While there may be value to having distinct Cupertino widgets, that's only if both Cupertino and Material see equal amounts of support.

Currently, Material widgets keep getting new API features and better accessibility updates that very slowly (if ever) trickle down to the Cupertino versions.

If the distinctions are mostly visual styling, it might make more sense to have the Cupertino version be a child of the Material widget.

The primary distinction that I can currently think of is that Cupertino widgets use the `CupertinoTheme` (which like the widgets has lagged behind the Material counterpart... especially with respect to dark mode). With respect to the `CupertinoTheme`, wouldn't it possibly make more sense to provide a `ColorScheme` that replicates the Cupertino light and dark colors?

**Comment by justinmc** [comment]
FYI @dkwingsmt for your work on the Cupertino API. It's probably already on your radar, but it would be nice to make the TextField and CupertinoTextField APIs match more closely.
